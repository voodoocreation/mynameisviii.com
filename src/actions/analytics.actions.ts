import actionCreatorFactory from "typescript-fsa";

const createAction = actionCreatorFactory("ANALYTICS");

export const trackEvent = createAction<{
  event: string;
  [index: string]: any;
}>("TRACK_EVENT");
