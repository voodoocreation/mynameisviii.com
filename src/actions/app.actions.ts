import actionCreatorFactory from "typescript-fsa";

const createAction = actionCreatorFactory("APP");

export const initApp = createAction.async<{ locale?: string }, {}>("INIT");

export const toggleNavigation = createAction("TOGGLE_NAVIGATION");

export const setOnlineStatus = createAction<boolean>("SET_ONLINE_STATUS");

export const setHasNewVersion = createAction<boolean>("SET_HAS_NEW_VERSION");

export const setCurrentRoute = createAction<string>("SET_CURRENT_ROUTE");

export const changeRoute = createAction.async<string, {}>("CHANGE_ROUTE");
