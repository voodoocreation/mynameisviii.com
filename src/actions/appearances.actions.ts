import actionCreatorFactory from "typescript-fsa";

import { IAppearance, IDynamoResponse, ILatLng } from "../models/root.models";

const createAction = actionCreatorFactory("APPEARANCES");

export const fetchAppearances = createAction.async<
  {},
  IDynamoResponse<IAppearance, "startingAt">
>("FETCH");

export const fetchMoreAppearances = createAction.async<
  {},
  IDynamoResponse<IAppearance, "startingAt">
>("FETCH_MORE");

export const fetchAppearanceBySlug = createAction.async<string, IAppearance>(
  "FETCH_BY_SLUG"
);

export const setCurrentAppearanceSlug = createAction<string>(
  "SET_CURRENT_SLUG"
);

export const geocodeCurrentAppearanceAddress = createAction.async<{}, ILatLng>(
  "GEOCODE_CURRENT_ADDRESS"
);
