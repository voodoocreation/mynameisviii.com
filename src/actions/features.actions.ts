import actionCreatorFactory from "typescript-fsa";

const actionCreator = actionCreatorFactory("FEATURES");

export const addFeatures = actionCreator<PLAddFeatures>("ADD");

export const removeFeatures = actionCreator<PLRemoveFeatures>("REMOVE");
