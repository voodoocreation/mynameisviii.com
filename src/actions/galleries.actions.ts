import actionCreatorFactory from "typescript-fsa";

const actionCreator = actionCreatorFactory("GALLERIES");

export const fetchGalleries = actionCreator.async<
  {},
  PLFetchGalleriesDone,
  PLFetchGalleriesFailed
>("FETCH");

export const fetchMoreGalleries = actionCreator.async<
  {},
  PLFetchGalleriesDone,
  PLFetchGalleriesFailed
>("FETCH_MORE");

export const fetchGalleryBySlug = actionCreator.async<
  PLFetchGalleryBySlugStarted,
  PLFetchGalleryBySlugDone,
  PLFetchGalleryBySlugFailed
>("FETCH_BY_SLUG");

export const setCurrentGallerySlug = actionCreator<PLSetCurrentGallerySlug>(
  "SET_CURRENT_SLUG"
);
