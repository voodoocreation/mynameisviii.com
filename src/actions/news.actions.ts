import actionCreatorFactory from "typescript-fsa";

const actionCreator = actionCreatorFactory("NEWS");

export const fetchLatestNews = actionCreator.async<
  {},
  PLFetchLatestNewsDone,
  PLFetchLatestNewsFailed
>("FETCH");

export const fetchMoreLatestNews = actionCreator.async<
  {},
  PLFetchLatestNewsDone,
  PLFetchLatestNewsFailed
>("FETCH_MORE");

export const fetchNewsArticleBySlug = actionCreator.async<
  PLFetchNewsArticleBySlugStarted,
  PLFetchNewsArticleBySlugDone,
  PLFetchNewsArticleBySlugFailed
>("FETCH_BY_SLUG");

export const setCurrentNewsArticleSlug = actionCreator<
  PLSetCurrentNewsArticleSlug
>("SET_CURRENT_SLUG");
