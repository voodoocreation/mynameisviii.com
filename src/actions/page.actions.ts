import actionCreatorFactory from "typescript-fsa";

const actionCreator = actionCreatorFactory("PAGE");

export const toggleNavigation = actionCreator<{}>("TOGGLE_NAVIGATION");

export const changeRoute = actionCreator.async<
  PLChangeRouteStarted,
  PLChangeRouteDone,
  PLChangeRouteFailed
>("CHANGE_ROUTE");
