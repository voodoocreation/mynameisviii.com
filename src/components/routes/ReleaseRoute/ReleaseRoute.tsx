import Head from "next/head";
import * as React from "react";
import { WrappedComponentProps } from "react-intl";
import { connect } from "react-redux";
import { bindActionCreators, Dispatch } from "redux";
import stripTags from "striptags";

import { absoluteUrl } from "../../../helpers/dataTransformers";
import injectIntlIntoPage from "../../../helpers/injectIntlIntoPage";
import { IRelease, IReleaseTrack } from "../../../models/root.models";
import { TStoreState } from "../../../reducers/root.reducers";
import { IPageContext } from "../../connected/App/App";
import Loader from "../../presentation/Loader/Loader";
import Release from "../../presentation/Release/Release";

import * as actions from "../../../actions/root.actions";
import * as selectors from "../../../selectors/root.selectors";

interface IProps extends WrappedComponentProps {
  fetchReleaseBySlug: typeof actions.fetchReleaseBySlug.started;
  isLoading: boolean;
  release?: IRelease;
  trackEvent: typeof actions.trackEvent;
}

class ReleaseRoute extends React.Component<IProps> {
  public static async getInitialProps(context: IPageContext) {
    const { query, store } = context;
    const slug = query.slug as string;

    store.dispatch(actions.setCurrentReleaseSlug(slug));

    const state = store.getState();

    if (!selectors.getCurrentRelease(state)) {
      store.dispatch(actions.fetchReleaseBySlug.started(slug));
    }
  }

  public render() {
    const { isLoading, release } = this.props;
    const { formatMessage } = this.props.intl;

    if (isLoading) {
      return <Loader className="PageLoader" />;
    }

    if (!release) {
      return null;
    }

    return (
      <>
        <Head>
          <title>
            {release.title}
            {" Â· "}
            {formatMessage({ id: "BRAND_NAME" })}
          </title>

          <meta
            content={stripTags(release.description).replace(/\n/g, "")}
            name="description"
          />

          <meta property="og:title" content={release.title} />
          <meta
            property="og:description"
            content={stripTags(release.description).replace(/\n/g, "")}
          />
          <meta
            property="og:url"
            content={absoluteUrl(`/releases/${release.slug}`)}
          />
          <meta property="og:type" content="music.album" />
          <meta property="music:release_date" content={release.releasedOn} />
          <meta property="music:musician" content={release.artist.url} />
          {release.tracklist.map(this.renderTracklistAlbumMeta)}
          <meta property="og:image" content={release.images[0].imageUrl} />
          <meta property="og:image:width" content="640" />
          <meta property="og:image:height" content="640" />
        </Head>

        <Release
          {...release}
          onCarouselSlideChange={this.onCarouselSlideChange}
        />
      </>
    );
  }

  private renderTracklistAlbumMeta = (
    album: IReleaseTrack[],
    albumIndex: number
  ) => (
    <React.Fragment key={albumIndex}>
      {album.map((track, trackIndex) => (
        <React.Fragment key={trackIndex}>
          <meta property="music:song" content={track.url} />
          <meta property="music:disc" content={`${albumIndex + 1}`} />
          <meta property="music:track" content={`${trackIndex + 1}`} />
        </React.Fragment>
      ))}
    </React.Fragment>
  );

  private onCarouselSlideChange = (index: number) => {
    this.props.trackEvent({
      event: "release.carousel.slideChange",
      index
    });
  };
}

const mapState = (state: TStoreState) => ({
  isLoading: selectors.getReleasesIsLoading(state),
  release: selectors.getCurrentRelease(state)
});

const mapActions = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      fetchReleaseBySlug: actions.fetchReleaseBySlug.started,
      trackEvent: actions.trackEvent
    },
    dispatch
  );

export default injectIntlIntoPage(connect(mapState, mapActions)(ReleaseRoute));
