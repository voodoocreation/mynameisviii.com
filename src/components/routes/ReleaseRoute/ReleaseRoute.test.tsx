import * as actions from "../../../actions/root.actions";
import { release } from "../../../models/root.models";
import MockPageContext from "../../../utilities/MockPageContext";
import WrapperWithRedux from "../../../utilities/WrapperWithRedux";
import ReleaseRoute from "./ReleaseRoute";

const item = release({
  images: [{ imageUrl: "Image URL 1" }, { imageUrl: "Image URL 2" }],
  slug: "test-1",
  tracklist: [[{ title: "Disc 1, Track 1" }], [{ title: "Disc 2, Track 1" }]],
});

const defaultState = {
  releases: {
    currentSlug: item.slug,
    items: {
      [item.slug]: item,
    },
  },
};

const component = new WrapperWithRedux(ReleaseRoute);

describe("[routes] <ReleaseRoute />", () => {
  describe("getInitialProps", () => {
    const context = new MockPageContext()
      .withDefaultQuery({
        slug: item.slug,
      })
      .withDefaultReduxState(defaultState);

    describe("when the release isn't already in the store", () => {
      const slug = "new-slug";

      it("calls getInitialProps method", async () => {
        await ReleaseRoute.getInitialProps(
          context.withQuery({ slug }).toObject()
        );
      });

      it("dispatches actions.setCurrentReleaseSlug with expected payload", () => {
        const matchingActions = context.reduxHistory.filter(
          actions.setCurrentReleaseSlug.match
        );

        expect(matchingActions).toHaveLength(1);
        expect(matchingActions[0].payload).toBe(slug);
      });

      it("dispatches actions.fetchReleaseBySlug.started", () => {
        expect(
          context.reduxHistory.filter(actions.fetchReleaseBySlug.started.match)
        ).toHaveLength(1);
      });
    });

    describe("when the article is already in the store", () => {
      it("calls getInitialProps method", async () => {
        await ReleaseRoute.getInitialProps(context.toObject());
      });

      it("dispatches actions.setCurrentReleaseSlug", () => {
        expect(
          context.reduxHistory.filter(actions.setCurrentReleaseSlug.match)
        ).toHaveLength(1);
      });

      it("doesn't dispatch actions.fetchReleaseBySlug.started", () => {
        expect(
          context.reduxHistory.filter(actions.fetchReleaseBySlug.started.match)
        ).toHaveLength(0);
      });
    });
  });

  it("matches snapshot", () => {
    const wrapper = component.withReduxState(defaultState).mount();

    expect(wrapper.render()).toMatchSnapshot();
  });

  it("renders a loader when isLoading is true", () => {
    const wrapper = component
      .withReduxState({
        ...defaultState,
        releases: { isLoading: true },
      })
      .mount();

    expect(wrapper.find("Loader")).toHaveLength(1);
  });

  it("doesn't render anything when no release exists", () => {
    const wrapper = component
      .withReduxState({
        news: {
          currentSlug: undefined,
          hasError: true,
          items: {},
        },
      })
      .mount();

    expect(wrapper.render().html()).toBeNull();
  });

  it("tracks carousel interactions correctly", () => {
    const wrapper = component.withReduxState(defaultState).mount();

    wrapper.find(".Carousel--pagination--page").last().simulate("click");

    const matchingActions = component.reduxHistory.filter(
      actions.trackEvent.match
    );

    expect(matchingActions).toHaveLength(1);
    expect(matchingActions[0].payload).toEqual({
      event: "release.carousel.slideChange",
      index: 1,
    });
  });
});
