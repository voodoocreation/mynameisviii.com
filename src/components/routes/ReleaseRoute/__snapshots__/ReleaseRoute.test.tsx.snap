// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[routes] <ReleaseRoute /> matches snapshot 1`] = `
<article
  class="Release"
>
  <header
    class="PageHeader"
  >
    <h1 />
  </header>
  <div
    class="MetaBar Release--meta"
  >
    <div
      class="Meta Release--type"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 14.5c-2.49 0-4.5-2.01-4.5-4.5S9.51 7.5 12 7.5s4.5 2.01 4.5 4.5-2.01 4.5-4.5 4.5zm0-5.5c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1z"
        />
      </svg>
       
      <span
        class="Meta--label"
      >
        Type:
      </span>
       
      <span
        class="Meta--value"
      >
        Album
      </span>
    </div>
    <div
      class="Meta Release--releasedOn"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
        />
      </svg>
       
      <span
        class="Meta--label"
      >
        Released:
      </span>
       
      <span
        class="Meta--value"
      >
        <time
          class="DateTime"
          datetime="2018-01-01"
          title="1 January 2018"
        >
          1 January 2018
        </time>
      </span>
    </div>
    <div
      class="Meta Release--length"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z"
        />
      </svg>
       
      <span
        class="Meta--label"
      >
        Length:
      </span>
       
      <span
        class="Meta--value"
      >
        0:00
      </span>
    </div>
    <div
      class="Meta Release--genre"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 3v10.55c-.59-.34-1.27-.55-2-.55-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4V7h4V3h-6z"
        />
      </svg>
       
      <span
        class="Meta--label"
      >
        Genre:
      </span>
       
      <span
        class="Meta--value"
      />
    </div>
    <div
      class="Meta Release--tracks"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 17h2v.5H3v1h1v.5H2v1h3v-4H2v1zm1-9h1V4H2v1h1v3zm-1 3h1.8L2 13.1v.9h3v-1H3.2L5 10.9V10H2v1zm5-6v2h14V5H7zm0 14h14v-2H7v2zm0-6h14v-2H7v2z"
        />
      </svg>
        
      <span
        class="Meta--value"
      >
        2 tracks
      </span>
    </div>
  </div>
  <div
    class="Release--body"
  >
    <section
      class="Release--images"
    >
      <div
        class="Carousel"
      >
        <div
          class="Carousel--slides"
          style="transform: translate3d(-0%, 0, 0);"
        >
          <figure
            class="Image Release--images--image isLoading"
          >
            <div
              class="Image--wrapper"
            >
              <img
                alt=""
                src="Image URL 1"
              />
              <span
                class="Loader"
              >
                <div
                  class="Loader--animation"
                >
                  <span />
                  <span />
                  <span />
                </div>
              </span>
            </div>
          </figure>
          <figure
            class="Image Release--images--image isLoading"
          >
            <div
              class="Image--wrapper"
            >
              <img
                alt=""
                src="Image URL 2"
              />
              <span
                class="Loader"
              >
                <div
                  class="Loader--animation"
                >
                  <span />
                  <span />
                  <span />
                </div>
              </span>
            </div>
          </figure>
        </div>
        <div
          class="Carousel--pagination"
        >
          <button
            class="Button Carousel--pagination--page isSelected"
            type="button"
          >
            <span
              class="Button--content"
            >
              <span>
                1
              </span>
            </span>
          </button>
          <button
            class="Button Carousel--pagination--page"
            type="button"
          >
            <span
              class="Button--content"
            >
              <span>
                2
              </span>
            </span>
          </button>
        </div>
      </div>
    </section>
    <div
      class="Release--details"
    >
      <section
        class="Release--description"
      />
      <section
        class="Release--tracklist"
      >
        <h2>
          Tracklist
        </h2>
        <h3>
          Disc 0
        </h3>
        <ol>
          <li>
            <span
              class="Release--tracklist--title"
            >
              Disc 1, Track 1
            </span>
             
            <span
              class="Release--tracklist--genre"
            />
             
            <span
              class="Release--tracklist--length"
            >
              0:00
            </span>
          </li>
        </ol>
        <h3>
          Disc 1
        </h3>
        <ol>
          <li>
            <span
              class="Release--tracklist--title"
            >
              Disc 2, Track 1
            </span>
             
            <span
              class="Release--tracklist--genre"
            />
             
            <span
              class="Release--tracklist--length"
            >
              0:00
            </span>
          </li>
        </ol>
      </section>
      <section
        class="Release--copyright"
      >
        <p>
          ℗© 2018 
        </p>
      </section>
    </div>
  </div>
  <script
    type="application/ld+json"
  >
    {"@id":"http://localhost:/releases/test-1","@type":"MusicAlbum","albumProductionType":"http://schema.org/StudioAlbum","albumRelease":{"@type":"MusicRelease","duration":"PT0M0S","name":"","recordLabel":""},"albumReleaseType":"http://schema.org/album","byArtist":{"@type":"MusicGroup","name":""},"datePublished":"2017-12-31T11:00:00.000Z","description":"","image":"Image URL 1","mainEntityOfPage":{"@id":"http://localhost:/releases/test-1"},"name":"","track":{"@type":"ItemList","itemListElement":[{"@type":"ListItem","item":{"@type":"MusicRecording","duration":"PT0M0S","name":"Disc 1, Track 1","recordingOf":{"@type":"MusicComposition","name":"Disc 1, Track 1"}},"position":1},{"@type":"ListItem","item":{"@type":"MusicRecording","duration":"PT0M0S","name":"Disc 2, Track 1","recordingOf":{"@type":"MusicComposition","name":"Disc 2, Track 1"}},"position":2}],"numberOfItems":2},"@context":"http://schema.org/"}
  </script>
</article>
`;
